packages/@uppy/provider-views/src/Browser.jsx

```diff
@@ -101,10 +101,16 @@ function Browser (props) {
               })}
 
               {files.map((file) => {
-                const validated = validateRestrictions(
-                  remoteFileObjToLocal(file),
-                  [...uppyFiles, ...currentSelection],
-                )
+                let validated
+                try {
+                  validateRestrictions(
+                    remoteFileObjToLocal(file),
+                    [...uppyFiles, ...currentSelection],
+                  )
+                  validated = { result: true }
+                } catch (err) {
+                  validated = { result: false, reason: err.message }
+                }
```

Comment:
You are just moving the strange API down 😅 

The idea is that we don't need a boolean, we only have an error variable that is undefined or defined. Then change everything that consumes this to check whether `err` is defined instead of `result:true`.

Two options:

1. Don't throw but return error in `restricter.validate`. Then we have `undefined` or `RestrictionError` as a value to deal with

```js
                const err = validateRestrictions(
                    remoteFileObjToLocal(file),
                    [...uppyFiles, ...currentSelection],
                  )
```

2. Keep `restricter.validate` like it is, but then you need to `try`/`catch` wrap it always with a let

Reply:
Yes, but what does it matter if anyone can use the actual error from Core? The public API is a real error like you wanted. I can get rid of the `result` thing and just check for the error, yeah.

Reply:
We should change our code too, yes. The question is which approach of the two mentioned above we should take?

Reply:
Updated to use the `restrictionError` directly.

Reply:
I'm still considering that the first option may be better. If the expected value is an error, not an unexpected error, than perhaps we don't need to 'blow up' the program with `throw`. We want the value after all. What do you think @aduh95?

Reply:
I don’t know if it’s still true, but there was a time where doing try/catch was much worse for perf in comparison with a nullish check (in V8 at least). Based on that, I’d say option 1. looks better.

Reply:
We can change the method name so the behavior is clearer (e.g. `getRestrictionError` or something)

Reply:
I wouldn't base it on what I'd call imaginary performance issues, as in, such things will never be the bottleneck in any web app. I also prefer `validateRestrictions` still. The type will become clear in JSDoc and/or TS anyway. But let's go for option 1 then :)

Reply:
So more or less same, but return instead of throw, so no need for weird try/catch. Ok.

Reply:
Might be nice to make things consistent and take along `validateMinNumberOfFiles` too

Reply:
validateMinNumberOfFiles is internal, only used in Core so far, and I didn't feel like changing just one check in Restricter, for consistency we’d have to change all of them to return?